src/main/scala/com.snowplowanalytics.sauna/apis/Slack.scala

```diff
@@ -0,0 +1,180 @@
+  def sendMessage(message: WebhookMessage): Future[WSResponse] = {
+    wsClient.url(webhookUrl)
+      .withHeaders("Content-Type" -> "application/json")
+      .post(Json.toJson(message))
+  }
+}
+
+object Slack {
+
+  /**
+   * Represents a Slack message that can be accepted by an incoming webhook. All
+   * elements are optional - some additional validation is required, but delegated to
+   * JSON schema validation.
+   *
+   * @see [[https://api.slack.com/docs/messages Slack API reference]]
+   */
+  case class WebhookMessage(
+    text: Option[String],
+    username: Option[String],
+    channel: Option[String],
+    iconUrl: Option[String],
+    iconEmoji: Option[String],
+    linkNames: Option[Boolean],
+    mrkdwn: Option[Boolean],
+    unfurlMedia: Option[Boolean],
+    unfurlLinks: Option[Boolean],
+    attachments: Option[Vector[WebhookAttachment]]
+  )
+
+  /**
+   * Represents a Slack message attachment.
+   *
+   * @see [[https://api.slack.com/docs/message-attachments Slack API reference]]
+   */
+  case class WebhookAttachment(
+    fallback: String,
+    color: Option[String],
+    pretext: Option[String],
+    authorName: Option[String],
+    authorLink: Option[String],
+    title: Option[String],
+    titleLink: Option[String],
+    text: Option[String],
+    fields: Option[Vector[WebhookAttachmentField]],
+    imageUrl: Option[String],
+    thumbUrl: Option[String],
+    footer: Option[String],
+    footerIcon: Option[String],
+    ts: Option[Int]
+  )
+
+  /**
+   * Represents a Slack attachment's field - a short key/value pair
+   * that will be displayed as a table cell in the attachment.
+   *
+   * @see [[https://api.slack.com/docs/message-attachments#fields Slack API reference]]
+   */
+  case class WebhookAttachmentField(
+    title: Option[String],
+    value: Option[String],
+    short: Option[Boolean]
+  )
+
+  implicit val webhookAttachmentFieldReads: Reads[WebhookAttachmentField] = (
+    (JsPath \ "title").readNullable[String] and
+      (JsPath \ "value").readNullable[String] and
+      (JsPath \ "short").readNullable[Boolean]
+    ) (WebhookAttachmentField.apply _)
+
+  implicit val webhookAttachmentReads: Reads[WebhookAttachment] = (
+    (JsPath \ "fallback").read[String] and
+      (JsPath \ "color").readNullable[String] and
+      (JsPath \ "pretext").readNullable[String] and
+      (JsPath \ "author_name").readNullable[String] and
+      (JsPath \ "author_link").readNullable[String] and
+      (JsPath \ "title").readNullable[String] and
+      (JsPath \ "title_link").readNullable[String] and
+      (JsPath \ "text").readNullable[String] and
+      (JsPath \ "fields").readNullable[Vector[WebhookAttachmentField]] and
+      (JsPath \ "image_url").readNullable[String] and
+      (JsPath \ "thumb_url").readNullable[String] and
+      (JsPath \ "footer").readNullable[String] and
+      (JsPath \ "footer_icon").readNullable[String] and
+      (JsPath \ "ts").readNullable[Int]
+    ) (WebhookAttachment.apply _)
```

Comment:
Done (though this also required rearranging the classes themselves to prevent Play JSON issues)